Reference/Events/Troubleshooting-Slow-Startup.md

```diff
@@ -0,0 +1,30 @@
+# Troubleshooting slow startup
+
+Slow startup can occur if: 
+* no indexes exist
+* models builder is used and is using Live modes and it's the first time it's compiling (you can use DLL mode to avoid this entirely)
+* the umbraco cache file doesn't exist
+* the `/App_Data/TEMP/PluginCache` is empty or if one of your DLLs has changed. This means a new scan of plugins will occur. Plugin scanning has also been heavily optimized since v7.5 (IIRC)
+* packages that do stuff on startup that don't need to 
+* custom code that do things on startup that don't need to
+
+_Troubleshooting tip: You can change your log4net config to *output Debug* level logs, this will give you detailed timing information on almost all aspects of startup so you can diagnose what might be running slow._
+
+# The request process
+
+Once booting is complete, then the request process occurs which is the [normal routing](../Routing/) phase.
+
+On first request the only thing that would be slower than the second request is resolving the URL to the published content item which is subsequently cached. I don't think this is a bottleneck though.
+
+# The rendering process
+
+Once the request process is complete, then the controller for the route will be launched followed by the rendering of the razor views.
+
+If your rendering is slow, be sure to review which controller is used. By default the [RenderMvcController](../../Implementation/default-routing/Controller-Selection/) will be used.  But it is possible to implement [custom routing](../Implementation/Custom-Routing/) to have bad performing code in a custom controller.
```

Comment:
or indeed replace the default RenderMvcController at startup with a custom RenderMvcController that may contain bad performing code....

Reply:
removed the word "indeed"

> Or the default RenderMvcController is replaced at startup with a custom RenderMvcController that may contain bad performing code.